crates/ruff_linter/src/rules/pylint/rules/unnecessary_dunder_call.rs

```diff
@@ -156,8 +157,45 @@ pub(crate) fn unnecessary_dunder_call(checker: &mut Checker, call: &ast::ExprCal
         call.range(),
     );
 
-    if let Some(fixed) = fixed {
-        diagnostic.set_fix(Fix::safe_edit(Edit::range_replacement(fixed, call.range())));
+    if let Some(mut fixed) = fixed {
+        let is_in_unary = checker
```

Comment:
The section here could use a comment explaining on why we're doing what we're doing below because it's not immediately obvious what it is about.